Tabloid/Repositories/PostRepository.cs

```diff
@@ -70,6 +70,66 @@ public List<Post> GetAll()
                 }
             }
         }
+        public Post GetPostById(int id)
+        {
+            using (var conn = Connection)
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"SELECT p.Title, p.Content, p.ImageLocation, p.CreateDateTime, p.PublishDateTime, p.IsApproved, p.CategoryId, p.UserProfileId,
+
+                                          up.Id AS UserId, up.FirebaseUserId, up.DisplayName, up.FirstName, up.LastName, up.Email, up.CreateDateTime AS UserCreateDateTime, up.ImageLocation, up.UserTypeId,
+
+                                        ut.Id AS TypeId, ut.Name AS TypeName
+
+                                          FROM Post p
+                                          JOIN UserProfile up ON p.UserProfileId = up.Id
+                                          JOIN UserType ut ON up.UserTypeId = ut.Id
+                                               WHERE p.Id = @Id";
+
+                    DbUtils.AddParameter(cmd, "@Id", id);
+
+                    using (SqlDataReader reader = cmd.ExecuteReader())
+                    {
+                        Post post = null;
+                        while (reader.Read())
+                        {
+                            post = new Post()
+                            {
+                                Id = id,
+                                Title = DbUtils.GetString(reader, "Title"),
+                                Content = DbUtils.GetString(reader, "Content"),
+                                ImageLocation = DbUtils.GetString(reader, "ImageLocation"),
+                                CreateDateTime = DbUtils.GetDateTime(reader, "CreateDateTime"),
+                                PublishDateTime = DbUtils.GetDateTime(reader, "PublishDateTime"),
+                                IsApproved = reader.GetBoolean(reader.GetOrdinal("IsApproved")),
```

Comment:
If there's no DbUtils method to support getting a bool, maybe you want to add one?

Reply:
I have added a new method to handle bools DbUtils.GetBool() and replaced them in PostRepository.cs